src/main/resources/com/puppycrawl/tools/checkstyle/sarif/SarifReport.template

```diff
@@ -11,7 +11,7 @@
           "language": "en",
           "name": "Checkstyle",
           "organization": "Checkstyle",
-          "rules": [
+          "rules": [${rules}
```

Comment:
Please put rules template on new line, it will always be big.

Reply:
If we do that, in case of no violations the JSON will look like this:
```
"rules": [
        
          ],
```
(note the blank line). 
Personally I do not think this looks nice and consistent with other elements in the report.

Current implementation handles both no-rules and present rules with consistent formatting of the final report (please see the test samples).

Reply:
Unfortunately, I have no time to satisfy subjective requirements. This project has been the most demanding (and not in all instances in a good sense) from all where I contributed over the last years. Please feel free to adjust the code per your liking or close the pull request.